lib/uplink/project.go

```diff
@@ -164,7 +165,7 @@ func (p *Project) OpenBucket(ctx context.Context, bucketName string, access *Enc
 	}
 	encryptionScheme := cfg.EncryptionParameters.ToEncryptionScheme()
 
-	ec := ecclient.NewClient(p.tc, p.uplinkCfg.Volatile.MaxMemory.Int())
+	ec := ecclient.NewClient(zap.L(), p.tc, p.uplinkCfg.Volatile.MaxMemory.Int())
```

Comment:
This should be passed down from somewhere.

Reply:
should we create a `zap.L()` in `cmd/uplink/cmd/root.go` and pass it down from there? As far as I know, there is no zap logger created on the uplink side right now

Reply:
I cannot find a way to pass it in from a higher level without needing to add `zap.L()` in more places. I could add it to the `libuplink.Uplink` struct and pass it down into the `Project` struct that way, but then, everywhere that calls `NewUplink` which is a lot of places will need to pass in `zap.L()`. Most places that call `NewUplink` do not have loggers already created. Should I still create them and pass them in from a higher level? I can do it but I don't think it's very pretty either way.

Reply:
if you want libuplink to do logging, i think *Uplink should not take a new constructor argument but instead should have a SetLogger method

Reply:
the reason is because most users won't care, and this is our public api

Reply:
I actually don't care about uplink logging at all in this case. I specifically care about ecclient logs on the satellite. Unfortunately, I can't add a logger to ecclient without having to handle it on the uplink side as well. And it can't just be `nil` otherwise we deal with lots of exceptions. So it doesn't seem like the `SetLogger` method is a solution for this case

Reply:
For uplink there probably should be two constructors. `New` and `NewWithLogger`, where `New` defaults to some `zap.L()`.

Reply:
SetLogger doesn't work?

```
ul, err := NewUplink(ctx, cfg)
ul.SetLogger(log)
// do stuff
```

Reply:
What I meant is - what if someone creates a new uplink and does not set a logger? Then in all the places where we use `uplink.logger.Debug` or something, we will get a nil pointer exception. 
Full disclosure, I do not know much about `SetLogger`, but I have implemented Egon's suggestion above, with `NewUplink` and `NewUplinkWithLogger`

Reply:
there's a zap.NewNop() that you would use by default.